kart/sqlalchemy/__init__.py

```diff
@@ -153,6 +153,26 @@ def separate_last_path_part(uri):
     return (modified_url, last_part)
 
 
+def truncate_to_path_length(uri, path_length):
+    """
+    Removes any number of path-parts from a URI to truncate it to the required length.
+    Eg, with path_length of 1, it would truncate SCHEME://HOST/DBNAME/DBSCHEMA/TABLE to SCHEME://HOST/DBNAME
```

Comment:
these right? looks like the two examples have been flipped

Reply:
feels like a test for this function specifically would be useful

Reply:
Dropped this function - realised "startswith" works pretty well here